ctx/ctx.go

```diff
@@ -114,45 +115,38 @@ func SetSession(r *http.Request, s *user.SessionState, scheduleUpdate bool, hash
 	}
 }
 
+// SetDefinition sets an API definition object to the request context.
 func SetDefinition(r *http.Request, s *apidef.APIDefinition) {
 	ctx := r.Context()
 	ctx = context.WithValue(ctx, Definition, s)
 	httputil.SetContext(r, ctx)
 }
 
+// GetDefinition will return a deep copy of the API definition valid for the request.
 func GetDefinition(r *http.Request) *apidef.APIDefinition {
 	if v := r.Context().Value(Definition); v != nil {
 		if val, ok := v.(*apidef.APIDefinition); ok {
-			return val
-		} else {
-			logger.Get().Warning("APIDefinition struct differ from the gateway version, trying to unmarshal.")
-			def := apidef.APIDefinition{}
-			b, _ := json.Marshal(v)
-			e := json.Unmarshal(b, &def)
-			if e == nil {
-				return &def
-			}
+			return reflect.Clone(val)
 		}
 	}
+
 	return nil
 }
 
-// GetOASDefinition returns a deep copy of the OAS definition of the called API.
-func GetOASDefinition(r *http.Request) *oas.OAS {
-	v := r.Context().Value(OASDefinition)
-	if v == nil {
-		return nil
-	}
-
-	val, ok := v.(*oas.OAS)
-	if !ok {
-		return nil
-	}
+// SetOASDefinition sets an OAS API definition object to the request context.
+func SetOASDefinition(r *http.Request, s *oas.OAS) {
+	ctx := r.Context()
```

Comment:
shall we use `httpctx.Value` here?

Reply:
Filed https://tyktech.atlassian.net/browse/TT-14063 to follow up, you can action the cleanup going forward. I'm happy with the delta here, no need to increase review surface.